File: SMART/app/src/main/java/com/cs565project/smart/db/entities/Category.java
package com.cs565project.smart.db.entities;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

@Entity
public class Category {
    @PrimaryKey
    @NonNull
    public String name;

    public boolean shouldRestrict;

    public Category(@NonNull String name, boolean shouldRestrict) {
        this.name = name;
        this.shouldRestrict = shouldRestrict;
    }
}

File: SMART/app/src/main/java/com/cs565project/smart/db/entities/Category.kt
package com.cs565project.smart.db.entities

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

/**
 * Database entity to hold categories of apps that user wants to restrict.
 */
@Entity
class Category(@field:PrimaryKey
               var name: String, var shouldRestrict: Boolean)
